src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql

```diff
@@ -1,43 +1,332 @@
+--
+--  PARAMETERS
+--      @version
+--          * The version number
+--      @status
+--          * The status of the version
+--
+CREATE PROCEDURE dbo.UpsertSchemaVersion
+    @version int,
+    @status varchar(10)
+AS
+    IF EXISTS(SELECT *
+        FROM dbo.SchemaVersion
+        WHERE Version = @version)
     BEGIN
         UPDATE dbo.SchemaVersion
-        SET [Status] = @status
-        WHERE [Version] = @version
+        SET Status = @status
+        WHERE Version = @version
     END
     ELSE
     BEGIN
-        INSERT INTO dbo.SchemaVersion ([Version], [Status])
-        VALUES (@version, @status)
+        INSERT INTO dbo.SchemaVersion
+            (Version, Status)
+        VALUES
+            (@version, @status)
     END
-END
+GO
+
+
+/*************************************************************
+    Model tables
+**************************************************************/
+
+CREATE TABLE dbo.SearchParam
+(
+    SearchParamId smallint IDENTITY(1,1) NOT NULL,
+    Uri varchar(128) NOT NULL
+)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_SearchParam ON dbo.SearchParam
+(
+    Uri
+)
+
+CREATE TABLE dbo.ResourceType
+(
+    ResourceTypeId smallint IDENTITY(1,1) NOT NULL,
+    Name nvarchar(50) NOT NULL
+)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_ResourceType on dbo.ResourceType
+(
+    Name
+)
+
+-- Create System and QuantityCode tables
+
+CREATE TABLE dbo.System
+(
+    SystemId int IDENTITY(1,1) NOT NULL,
+    System nvarchar(256) NOT NULL,
+)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_System ON dbo.System
+(
+    System
+)
+
+CREATE TABLE dbo.QuantityCode
+(
+    QuantityCodeId int IDENTITY(1,1) NOT NULL,
+    QuantityCode nvarchar(256) NOT NULL
+)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_QuantityCode on dbo.QuantityCode
+(
+    QuantityCode
+)
+
+/*************************************************************
+    Resource table
+**************************************************************/
+
+CREATE TABLE dbo.Resource
+(
+    ResourceTypeId smallint NOT NULL,
+    ResourceId varchar(64) NOT NULL,
+    Version int NOT NULL,
+    IsHistory bit NOT NULL,
+    ResourceSurrogateId bigint NOT NULL,
+    LastUpdated datetime2(7) NOT NULL,
+    IsDeleted bit NOT NULL,
+    RequestMethod varchar(10) NULL,
+    RawResource varbinary(max) NOT NULL
+)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_Resource ON dbo.Resource (
+    ResourceSurrogateId
+)
+
+CREATE UNIQUE NONCLUSTERED INDEX IX_Resource_ResourceTypeId_ResourceId_Version ON dbo.Resource (
+    ResourceTypeId,
+    ResourceId,
+    Version
+)
+
+CREATE UNIQUE INDEX IX_Resource_ResourceTypeId_ResourceId ON dbo.Resource (
+    ResourceTypeId,
+    ResourceId
+)
+INCLUDE (Version)
+WHERE IsHistory = 0
+
+GO
+
+CREATE SEQUENCE dbo.ResourceSurrograteIdSequence
+        AS BIGINT
+        START WITH 0
+        INCREMENT BY 1
+        NO CYCLE
+        CACHE 50
+GO
+
+/*************************************************************
+    Stored procedures for creating and deleting
```

Comment:
Should we set nocount on in our stored procedures?

Reply:
Yes, we should

Reply:
(9 bytes per statement, so not going to cause a big difference, but good to do)